amps/ags/rm-community/rm-community-repo/source/java/org/alfresco/module/org_alfresco_module_rm/disposition/DispositionService.java

```diff
@@ -288,4 +288,12 @@ DispositionActionDefinition updateDispositionActionDefinition(
      * @param record
      */
     void recalculateNextDispositionStep(NodeRef record);
+
+    /**
+     * Check given node or its children are frozen
+     * The node should be record or record folder for retention schedule
+     *
+     * @param nodeRef
+     */
+    boolean isFrozenOrHasFrozenChildren(NodeRef nodeRef);
```

Comment:
DispositionService is a public interface with the methods related to disposition/retention defined this method is related to freezing service...not clear to me why this should belong to this current public interface and not within FreezeService

Reply:
I thought the same thing - perhaps the HoldService would be an even better home for this method?